.github/workflows/tag-build.yaml

```diff
@@ -1,10 +1,220 @@
+                      ;;
+                  "push")
+                      REF_TYPE=$(echo "$GITHUB_EVENT_REF" | cut -d'/' -f2)
+                      # Case: commit push event.
+                      if [ "$REF_TYPE" != "tags" ]; then
+                          echo "environment=build-stage"
+                          echo "version=$(generate_version "$GITHUB_EVENT_REF_NAME")"
+                          return
+                      fi
+          
+                      # Case: tag event with prerelease version.
+                      if [ "${GITHUB_EVENT_REF_NAME#*-}" != "$GITHUB_EVENT_REF_NAME" ]; then
+                          echo "environment=build-stage"
+                          echo "version=$GITHUB_EVENT_REF_NAME"
+                          return
+                      fi
+          
+                      # Case: tag event with release version. Only this
+                      # should go to prod.
+                      echo "environment=build-prod"
+                      echo "version=$GITHUB_EVENT_REF_NAME"
+                      ;;
+                  *)
+                      >&2 echo "Unknown GHA event $GITHUB_EVENT_NAME, failing" 
+                      exit 1
+                      ;;
+              esac
+          }
+          
+          get_output_vars >> "$GITHUB_OUTPUT"
+          
+          # Validate the semver
+          . "$GITHUB_OUTPUT"  # Load the variables into the current environment
+          (echo "$version" | grep -qP "$SEMVER_REGEX") || (echo "The artifact version $version is not a valid semver-coerced value"; exit 1)
+        
+          # Log the build details
+          echo "Built config:" | tee -a "$GITHUB_STEP_SUMMARY"
+          cat "$GITHUB_OUTPUT" | while read line; do
+            echo "* $line" | tee -a "$GITHUB_STEP_SUMMARY"
+          done
+      - name: ${{ inputs.workflow-tag }}
+        if: inputs.workflow-tag != ''
+        run: |
+          # Do nothing
+  # Each section here could be split out into a separate job, at the cost of slightly increased complexity.
+  # This would improve the (already somewhat fast) performance a bit, but I'm not sure if it's worth the
+  # tradeoff.
+  build-plugins:
+    needs:
+      - setup
+    runs-on: ubuntu-22.04-32core
+    environment: ${{ needs.setup.outputs.environment }}
+    permissions:
+      contents: read
+      id-token: write
+    env:
+      ARTIFACT_DIRECTORY: /tmp/build
+    steps:
+      # Setup
+      - name: Enable performance telemetry/metrics
+        uses: catchpoint/workflow-telemetry-action@v1
+        with:
+          comment_on_pr: false
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.setup.outputs.gitref }}
+          fetch-depth: 0  # This is required by some of the commands in the makefiles
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "./go.mod"
+          cache-dependency-path: "./go.sum"
```

Comment:
Are you certain? Looking at the action code it appears that if not specified the action will try to use the pre-installed version: https://github.com/actions/setup-go/blob/main/src/main.ts#L57

You are right about the `cache-dependency-path` though.

Reply:
Did I accidentally select both lines? I meant to only comment on `cache-dependency-path`. I did originally select both because I thought the default for the version file would be that too, but double checked and realised that was wrong. I must have forgotten to change my comment selection. Sorry.

So, confirming that `go-version-file: "./go.mod"` is not redundant, just the `cache-dependency-path` parameter.